### DISCUSSION TITLE
[QUESTION] Can `strict-local` volumes be moved?

### DISCUSSION THREAD

**Discussion by jobo-prod** [longhorn/longhorn#6539] [General]
## Question

Hello everybody,
The `strict-local` data locality policy looks very promising to me. I think I might be migrating local volumes from Rancher's local-path-provisioner to Longhorn, since it offers a lot more functionality (snapshots, backups, etc.). The docs are a little bit short on this topic, but according to what I've been reading in docs and discussions, it seems like strict-local volumes cannot be moved between nodes, so:

- Is it true that I cannot move `strict-local` volumes, even if I want to do it manually and intentionally? I think this would be another big advantage over simpler solutions for local storage, because it really comes in handy when scaling clusters, for example.
- If yes, what is the best way to mograte strict-local storage to a different node?

My use-case is a kind of storage that is guaranteed "local" to the pod's location, but not necessarily tied to one specific node (since I could control that via pod scheduling constraints, anyway). It would be awesome to have the speed and overhead benefits of `strict-local` volumes that could still "follow" their workloads around the cluster, if necessary.

Thanks in advance!

## Environment

 - Longhorn version: 1.5.1
 - Kubernetes version: 1.22
 - Node config
   - OS type and version: Ubuntu 20
 - Underlying Infrastructure: hybrid vSphere/bare-metal

**Comment by weizhe0422** [comment]
I think yes that the `strict-local` volume couldn't attach to another node. Since both the engine and replica should be on the same node using Unix domain socket to interact, it is done to increase the performance. And I think the Pod will be scheduled to the node with replica even the Pod restart.

ref.
 - https://github.com/longhorn/longhorn/blob/master/enhancements/20221123-local-volume.md
 - https://github.com/longhorn/longhorn/issues/5025
 
cc @derekbit

**Reply by jobo-prod** [reply]
Thanks for the explanation, @weizhe0422! According to the first link:
```
A local volume's data locality cannot be converted to other modes when volume is not detached
```
This sounds to me like it is possible to convert a volume to another mode (when it is detached), which would be a solution to moving data from a `strict-local` volume elsewhere. Is that true?

**Reply by derekbit** [reply]
@bauerjs1 Correct!

**Reply by jobo-prod** [reply]
Awesome, thank you! 💯
